providers/implementations/keymgmt/dh_kmgmt.c

```diff
@@ -158,6 +164,44 @@ static int dh_exportdomparams(void *domparams, OSSL_CALLBACK *param_cb,
     return ret;
 }
 
+static int dh_isdomparams(const void *domparams)
+{
+    /*
+     * dh should always contain the domain parameters, so we could as well
+     * return 1 here and be done with it.  However, future development might
+     * change this, so we make this future proof and test for real.
+     */
+    return DH_get0_p(domparams) != NULL && DH_get0_g(domparams) != NULL;
+}
+
+static int dh_cmpdomparams(const void *domparams1, const void *domparams2)
+{
+    const BIGNUM *q1, *q2;
+
+    if (BN_cmp(DH_get0_p(domparams1), DH_get0_p(domparams2)) != 0
+        && BN_cmp(DH_get0_g(domparams1), DH_get0_g(domparams2)) != 0)
+        return 0;
+    /* Support DHX, compare Q if available */
+    q1 = DH_get0_q(domparams1);
+    q2 = DH_get0_q(domparams2);
+    if (q1 == NULL && q2 == NULL)
```

Comment:
Not sure why you need these null checks : BN_cmp returns 0 if both NULL or if they are equal otherwise it returns -1 or 1 dependent on a < b.. So the cmp()==0 should be sufficient.

Reply:
I hadn't looked at `BN_cmp`'s behavior. Thanks for the heads up